### PR TITLE
Update test_multiarray.py to support numpy 2.0+

### PR DESCRIPTION
**Pull request by ccoulombe**
Import _core instead of core.

Addresses partially #137182

### CODE REVIEW

File: test/torch_np/numpy_tests/core/test_multiarray.py [hunk 1]
```diff
@@ -75,7 +75,11 @@
 IS_PYSTON = False
 HAS_REFCOUNT = True
 
-from numpy.core.tests._locales import CommaDecimalPointLocale
+if numpy.__version__ > "2":
+    # numpy 2.0 +, see https://numpy.org/doc/stable/release/2.0.0-notes.html#renamed-numpy-core-to-numpy-core
+    from numpy._core.tests._locales import CommaDecimalPointLocale
+else:
+    from numpy.core.tests._locales import CommaDecimalPointLocale
```
Comment (bdhirsh) [inline test/torch_np/numpy_tests/core/test_multiarray.py:82]: @albanD who would be a good person to review numpy testing changes?

Comment (colesbury) [reply test/torch_np/numpy_tests/core/test_multiarray.py:82]: Alban is on vacation. Maybe @ezyang knows?

**Comment by ezyang** [comment]
need cla

**Review by ezyang — APPROVED** [review]

**Comment by ccoulombe** [comment]
> need cla

Now signed this one as well!

**Review by albanD — APPROVED** [review]
Sounds good!

**Comment by albanD** [comment]
Ho can you fix the lint please?

**Comment by ccoulombe** [comment]
> Ho can you fix the lint please?

sure will do by the EOW

**Pull request closed without merging**